### PR TITLE
 Make hack/validate* scripts use & require path arg 

### PR DESCRIPTION
**Pull request by petr-muller**
Followup to https://github.com/openshift/release/pull/7072. Make all `validate` scripts use the explicit path argument instead of relying on `$BASH_SOURCE` and/or `$CWD`. After this PR, openshift/release presubmits are expected to work normally, but the presubmits on https://github.com/redhat-operator-ecosystem/release that use these scripts should start to explicitly fail (and need to be fixed) where they previously could spuriously pass.

I also removed `validate-jenkinsfiles.sh` (unused) and fixed few minor code problems.

/cc @openshift/openshift-team-developer-productivity-test-platform

### CODE REVIEW

**Comment by petr-muller** [comment]
Really, good old friend pylint? Bugging me instead of Steve?

/meow

File: hack/order-prow-job-config.sh [hunk 1]
```diff
@@ -7,6 +7,13 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-ci_operator_dir="$( dirname "${BASH_SOURCE[0]}" )/../ci-operator"
+base_dir="${1-}"
+
+if [[ -z "${base_dir}" ]] || ! [[ -d "${base_dir}" ]]; then
```
Comment (stevekuznetsov) [inline hack/order-prow-job-config.sh]: nit: the point of using `[[` over `[` is that you can stay inside of one statement and not use bash logic along with `test` logic...


```sh
if [[ -z "${base_dir:-}" || ! -d "${base_dir}" ]]; then
```

Although you could really just get away with:

```sh
if [[ ! -d "${1:-}" ]]; then
```

Comment (petr-muller) [reply hack/order-prow-job-config.sh]: Thanks, addressed.

**Comment by petr-muller** [comment]
/hold

Found a problem, mmnt

**Comment by petr-muller** [comment]
/hold cancel

Fixed (I needed to keep one `$BASH_SOURCE` usage so taht one `hack/` script can call another).

https://github.com/openshift/release/compare/eee7973828f8efc75ebacc3db0d4ef8519c5380f..e39d7a56b1a50f838daf79262e20f8cc378c4ed0

**Comment by petr-muller** [comment]
@stevekuznetsov :pray: re-lgtm? I conflicted with Bruno's semantics check speedup.

**Pull request merged**